slsk-batchdl/Extractors/Apis/BandcampApi.cs

```diff
@@ -0,0 +1,193 @@
+using Models;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+
+namespace Extractors.Apis
+{
+    public static class BandcampApi
+    {
+        public static async Task<string?> ExtractFanIdFromPage(string wishlistUrl)
+        {
+            var web = new HtmlAgilityPack.HtmlWeb();
+            var doc = await web.LoadFromWebAsync(wishlistUrl);
+
+            // Extract fan_id from the page HTML
+            string fanIdPattern = @"fan_id[""']?\s*[:=]\s*(\d+)";
+            var pageContent = doc.DocumentNode.OuterHtml;
+            var fanIdMatch = Regex.Match(pageContent, fanIdPattern);
+
+            if (!fanIdMatch.Success)
+            {
+                Logger.Warn("Could not extract fan_id from page HTML");
+                return null;
+            }
+
+            string fanId = fanIdMatch.Groups[1].Value;
+            Logger.Info($"Found fan_id: {fanId}");
+            return fanId;
+        }
+
+
+
+        public static async Task FetchWishlistItemsFromAPI(string fanId, TrackLists trackLists, string fanUsername = "")
+        {
+            using var httpClient = new HttpClient();
+
+            string apiUrl = "https://bandcamp.com/api/fancollection/1/wishlist_items";
+            int startingItemCount = trackLists.lists.Count;
+            int itemCount = startingItemCount;
+            string? olderThanToken = null;
+            int batchSize = 20;
+
+            while (true)
+            {
+                try
+                {
+                    if (olderThanToken == null)
+                    {
+                        // Generate a token with current timestamp for the initial request
+                        long currentTimestamp = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
+                        olderThanToken = $"{currentTimestamp}:0:a::";
```

Comment:
this having `:0:` doesn't replicate the real request a browser makes, but still works. Not really sure what that number is in the API request but it works still...

Reply:
It's the ID of the last album in the initial html document, so it can probably stay as 0 but might look at seeing how hard it is to just parse it out of it as we already make the request.